ssl/t1_lib.c

```diff
@@ -2743,6 +2728,30 @@ int ssl_check_clienthello_tlsext_late(SSL *s)
     } else
         s->tlsext_status_expected = 0;
 
+    if (s->ctx->alpn_select_cb != NULL) {
+        const unsigned char *selected = NULL;
+        unsigned char selected_len = 0; /* yes, this is unsigned char */
```

Comment:
Right, but you added a comment: I'm just saying that the `unsigned char` isn't ridiculous (well, uint8_t would be better), because it ensures that the response is in the right range.